### PR TITLE
[stable/postgresql] Fix volume's hidden directories from crashing initdb

### LINKED ISSUE #415: postgresql chart fails when PVC is backed by glusterfs
When using glusterfs backed PVC, the gluster volumes automatically have two hidden directories: `.glusterfs` and `.trashcan`.  This causes the postgresql `initdb` task to fail because it wants an empty directory.  The postgresql pod will crash loop due to it.

Logs from postgresql container:

```
initdb: directory "/var/lib/postgresql/data/pgdata" exists but is not empty
It contains a dot-prefixed/invisible file, perhaps due to it being a mount point.
Using a mount point directly as the data directory is not recommended.
Create a subdirectory under the mount point.
```

### PR DESCRIPTION
**Pull request by ravishivt**
Fixes #415.

### CODE REVIEW

**Comment by ravishivt** [comment]
I signed it!

**Comment by ravishivt** [comment]
I switched the approach from mounting the data volume one directory up to using the volumeMount's [subPath feature](https://github.com/kubernetes/kubernetes/pull/22575).  Will leave it up for discussion on which is better but IMO, subPath is superior because it also removes the need for an init-container.

**Comment by viglesiasce** [comment]
Yeah I like that. I wonder what we can do to make this change backwards compatible or if we need to make this a breaking change.

cc @databus23

**Comment by databus23** [comment]
I think this will be a breaking change. Unless we include some kind of migration init-container that moves data around I don't see how it can't be. And I don't like the idea of shovelling around user data anyway.
Also if we go with the `subPath` option I think the container won't even be able to detect that there is a database  present in the root of the PVC let alone move it because the container only sees the empty subdirectory.

While I like going for sub directory and removing the init container this change still makes me a little uncomfortable because the "backward incompatibility" in this case is exceptionally bad. If users upgrade the postgres version and are not aware of this change, they will end up with an empty database setup from scratch in the subdirectory. That old data is not gone, it still sits in the root of the PVC but recovering from that is a hassle.

Maybe to at least give existing installations on option  we should make the `subPath` configurable via value. That way setting it to `""` would allow users to upgrade and keep running postgres in the root of the PVC.

**Comment by viglesiasce** [comment]
@databus23 that sounds like a good plan:

- Make subPath configurable
- Do a major version bump

We will need to do this for other charts too like Jenkins.

**Comment by viglesiasce** [comment]
@k8s-bot ok to test

**Comment by ravishivt** [comment]
OK the latest commit addresses the concerns.  Also, I bumped the version to 1.0.0.  Not sure if we wanted to instead go with 0.4.0.

@viglesiasce Does Jenkins have this issue as well?  I deployed Jenkins successfully without adding a subPath.  It doesn't try to remove `lost+found` in an init container.  That said, `mysql` does so it's another subPath target.

**Review by databus23 — APPROVED** [review]
lgtm

**Comment by lachie83** [comment]
@k8s-bot ok to test

**Comment by lachie83** [comment]
lgtm

File: stable/postgresql/README.md [hunk 1]
```diff
@@ -88,5 +89,7 @@ The [postgres](https://github.com/docker-library/postgres) image stores the Post
 
 The chart mounts a [Persistent Volume](http://kubernetes.io/docs/user-guide/persistent-volumes/) volume at this location. The volume is created using dynamic volume provisioning.
 
+The volume defaults to mount at a subdirectory of the volume instead of the volume root to avoid the volume's hidden directories from interfering with `initdb`.  If you are upgrading this chart from before version `1.x.x`, set `persistence.subPath` to `""`.
```
Comment (mgoodness) [inline stable/postgresql/README.md]: Change version reference from `1.x.x` to `0.4.0`

**Review by mgoodness — CHANGES_REQUESTED** [review]

**Comment by lachie83** [comment]
@mgoodness updated

**Review by mgoodness — APPROVED** [review]

**Pull request merged**